File: ios/ThreeDSecure.m
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(ThreeDSecure, RCTViewManager)
  RCT_EXTERN_METHOD(requestThreeDSecure: (NSDictionary*)parametres3DS resolve: (RCTPromiseResolveBlock)resolve reject: (RCTPromiseRejectBlock)reject)
@end

File: ios/ThreeDSecure.swift
import Foundation;

@objc(ThreeDSecure)
class ThreeDSecure: NSObject {
  public static var resolve: RCTPromiseResolveBlock?;
  public static var reject: RCTPromiseRejectBlock?;

  @objc
  public func requestThreeDSecure(_ parametres3DS: Dictionary<String, String>, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
    let transactionId = parametres3DS["transactionId"]!;
    let paReq = parametres3DS["paReq"]!;
    let acsUrl = parametres3DS["acsUrl"]!;

    ThreeDSecure.resolve = resolve;
    ThreeDSecure.reject = reject;

    DispatchQueue.main.async {
      let threeDSecure = ThreeDSecureController();

      threeDSecure.onShowController();
      threeDSecure.onRequest(transactionId: transactionId, paReq: paReq, acsUrl: acsUrl);
    }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
